clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -229,7 +229,7 @@ public long connectionDelay(Node node, long now) {
      */
     @Override
     public boolean connectionFailed(Node node) {
-        return connectionStates.connectionState(node.idString()).equals(ConnectionState.DISCONNECTED);
+        return connectionStates.isDisconnected(node.idString());
```

Comment:
I'm not sure about this. It's weird to say that a connection that has never been attempted has failed. This kind of situation only arises due to specifics of how `ConsumerNetworkClient` is implemented, so we should perhaps handle it there. cc @hachikuji